xcdat/dataset.py

```diff
@@ -337,211 +312,118 @@ def decode_non_cf_time(dataset: xr.Dataset) -> xr.Dataset:
-    # in the `.encoding` dict to mimic xarray's behavior.
-    calendar = time_attrs.pop("calendar", None)
-    units_attr = time_attrs.pop("units", None)
-
-    if calendar is None:
-        logger.warning(
-            "This dataset's time coordinates do not have a 'calendar' attribute set, "
-            "so time coordinates could not be decoded. Set the 'calendar' attribute "
-            f"(`ds.{time.name}.attrs['calendar]`) and try decoding the time "
-            "coordinates again."
-        )
-        return ds
-
-    if units_attr is None:
-        logger.warning(
-            "This dataset's time coordinates do not have a 'units' attribute set, "
-            "so the time coordinates could not be decoded. Set the 'units' attribute "
-            f"(`ds.{time.name}.attrs['units']`) and try decoding the time "
-            "coordinates again."
-        )
-        return ds
-
-    try:
-        units, ref_date = _split_time_units_attr(units_attr)
-    except ValueError:
-        logger.warning(
-            f"This dataset's time coordinates 'units' attribute ('{units_attr}') is "
-            "not in a supported format ('months since...' or 'years since...'), so the "
-            "time coordinates could not be decoded."
-        )
-        return ds
-
-    data = _get_cftime_coords(ref_date, time.values, calendar, units)
-    decoded_time = xr.DataArray(
-        name=time.name,
-        data=data,
-        dims=time.dims,
-        coords={time.name: data},
-        # As mentioned in a comment above, the units and calendar attributes are
-        # popped from the `.attrs` dict.
-        attrs=time_attrs,
-    )
-    decoded_time.encoding = {
-        "source": ds.encoding.get("source", "None"),
-        "dtype": time.dtype,
-        "original_shape": time.shape,
-        # The units and calendar attributes are now saved in the `.encoding`
-        # dict.
-        "units": units_attr,
-        "calendar": calendar,
-    }
-    ds = ds.assign_coords({time.name: decoded_time})
-
-    try:
-        time_bounds = ds.bounds.get_bounds("T")
-    except KeyError:
-        time_bounds = None
-
-    if time_bounds is not None:
-        lowers = _get_cftime_coords(ref_date, time_bounds.values[:, 0], calendar, units)
-        uppers = _get_cftime_coords(ref_date, time_bounds.values[:, 1], calendar, units)
-        data_bounds = np.vstack((lowers, uppers)).T
-
-        decoded_time_bnds = xr.DataArray(
-            name=time_bounds.name,
-            data=data_bounds,
-            dims=time_bounds.dims,
-            coords=time_bounds.coords,
-            attrs=time_bounds.attrs,
-        )
-        decoded_time_bnds.coords[time.name] = decoded_time
-        decoded_time_bnds.encoding = time_bounds.encoding
-        ds = ds.assign({time_bounds.name: decoded_time_bnds})
+    coord_keys = list(get_axis_coord(ds, "T").coords)
+
+    for key in coord_keys:
+        coord = ds[key].copy()
```

Comment:
Note that when I open the dataset below, I get this warning:

> 2022-10-12 16:56:28,048 [WARNING]: dataset.py(decode_time:333) >> height does not have a 'calendar' attribute set, so it could not be decoded. Set the 'calendar' attribute (`ds.height.attrs['calendar]`) and try decoding them again

I think we can resolve this by only decoding coords that are also dims, by adding the following at the top of this for loop: 

```
        if key not in ds.dims:
            continue
```

Reply:
This warning is now fixed.

Solution:
https://github.com/xCDAT/xcdat/blob/ad158e44a6094d369773fa405ac7d49cda82156b/xcdat/dataset.py#L315-L316

We reference all coordinate keys related to the time axis, rather than getting the dimension coordinates and referring to the `.coords` attr (which can have the unrelated singletons attached to it).